flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBListState.java

```diff
@@ -242,6 +244,36 @@ public void addAll(List<V> values) {
 		}
 	}
 
+	@Override
+	public void migrateSerializedValue(
+			DataInputDeserializer serializedOldValueInput,
+			DataOutputSerializer serializedMigratedValueOutput,
+			TypeSerializer<List<V>> priorSerializer,
+			TypeSerializer<List<V>> newSerializer) throws StateMigrationException {
+
+		Preconditions.checkArgument(priorSerializer instanceof ListSerializer);
+		Preconditions.checkArgument(newSerializer instanceof ListSerializer);
+
+		TypeSerializer<V> priorElementSerializer =
+			((ListSerializer<V>) priorSerializer).getElementSerializer();
+
+		TypeSerializer<V> newElementSerializer =
+			((ListSerializer<V>) newSerializer).getElementSerializer();
+
+		try {
+			while (serializedOldValueInput.available() > 0) {
```

Comment:
Why can't we use `deserializeNextElement` for the read part and instead duplicate some of the code?